pkg/helper/client.go

```diff
@@ -104,3 +115,30 @@ func ApplyDesiredState(nodeNetworkState *nmstatev1alpha1.NodeNetworkState) (stri
 	}
 	return set(string(nodeNetworkState.Spec.DesiredState))
 }
+
+func filterOut(currentState nmstatev1alpha1.State, interfacesFilter string) nmstatev1alpha1.State {
+	var state map[string]interface{}
```

Comment:
Add here the empty check
if interfacesFilter == "" {
    return currentState
}